File: docs/extensibility/internals/codesnippet/VisualBasic/creating-options-pages_3.vb
    Private Sub ShowHelloWorld(ByVal sender As Object, ByVal e As EventArgs)
        Dim hw = TryCast(GetDialogPage(GetType(HelloWorldOptions)), HelloWorldOptions)
        If hw.HelloWorld = True Then
            MessageBox.Show("Hello World!")
        Else
            MessageBox.Show("Goodbye.")
        End If
    End Sub

File: docs/extensibility/internals/codesnippet/CSharp/creating-options-pages_3.cs
        private void ShowHelloWorld(object sender, EventArgs e)
        {
            var hw = GetDialogPage(typeof(HelloWorldOptions)) 
                as HelloWorldOptions;
            if (hw.HelloWorld == true)
            {
                MessageBox.Show("Hello World!");
            }
            else MessageBox.Show("Goodbye.");
        }